examples/module/imu/imu_corrector.py

```diff
@@ -81,41 +102,88 @@ def test(network, loader, device = "cuda:0"):
-                              duration=10, mode='train', download=True)
+                              duration=args.duration, mode='train', download=True)
     test_dataset = KITTI_IMU(args.dataroot, args.dataname, args.datadrive[0],
-                             duration=10, mode='test', download=True)
+                             duration=args.duration, mode='test', download=True)
+    evaluate_dataset = KITTI_IMU(args.dataroot, args.dataname, args.datadrive[0],
+                            duration=10, step_size=10, mode='evaluate', download=True)
+
     train_loader = Data.DataLoader(dataset=train_dataset, batch_size=args.batch_size,
                                    collate_fn=imu_collate, shuffle=True)
     test_loader = Data.DataLoader(dataset=test_dataset, batch_size=args.batch_size,
                                   collate_fn=imu_collate, shuffle=False)
+    evaluate_loader = Data.DataLoader(dataset=evaluate_dataset, batch_size=1,
+                                      collate_fn=imu_collate, shuffle=False)
 
-    ## optimizer
     network = IMUCorrector().to(args.device)
-    optimizer = torch.optim.Adam(network.parameters(), lr=5e-6)  # to use with ViTs
-    scheduler = ReduceLROnPlateau(optimizer, 'min', factor=0.1, patience=10)
+    integrator = IMUIntegrator().to(args.device)
+
+    optimizer = torch.optim.Adam(network.parameters(), lr = 5e-5)
+    scheduler = ReduceLROnPlateau(optimizer, 'min', factor = 0.1, patience = 15)
 
-    for epoch in range(args.max_epoches):
-        train_loss = train(network, train_loader, epoch, optimizer, device = args.device)
+    best_loss = torch.inf
+    para_stae_dict = None
+    for epoch_i in range(args.max_epoches):
+        train_loss = train(network, train_loader, epoch_i, optimizer, device = args.device)
         test_loss = test(network, test_loader, device = args.device)
-        scheduler.step(train_loss)
-        print("train loss: %f test loss: %f "%(train_loss, test_loss))
+        reference_loss = test(integrator, train_loader, device = args.device)
+        scheduler.step(test_loss)
+        if best_loss > test_loss:
+            best_loss = test_loss
+            para_stae_dict = network.state_dict()
+        print("train loss: %f test loss: %f reference loss: %f "%(train_loss, test_loss, reference_loss))
+
+    print("evaluating...")
+
+    init = evaluate_dataset.get_init_value()
+    network = IMUCorrector().to(args.device)
+    network.load_state_dict(para_stae_dict)
+    poses_corrected, poses_gt = evaluate(network, init, evaluate_loader, device = args.device)
+    poses, _ = evaluate(integrator, init, evaluate_loader, device = args.device)
+
+    plt.figure(figsize=(5, 5))
+    ax = plt.axes()
+    ax.plot(poses[:,0], poses[:,1], 'b')
+    ax.plot(poses_corrected[:,0], poses_corrected[:,1], 'g')
+    ax.plot(poses_gt[:,0], poses_gt[:,1], 'r')
+    plt.title("PyPose IMU Integrator")
+    plt.legend(["Integration", "Corrected", "Ground Truth"])
+
+    plt.show()
```

Comment:
do you want a single plot function?

Reply:
Embed the plotting process into a plot function? @wang-chen

Reply:
yes,